protocols/delegate-manager/contracts/DelegateManager.sol

```diff
@@ -0,0 +1,120 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity 0.5.10;
+pragma experimental ABIEncoderV2;
+
+import "@airswap/indexer/contracts/interfaces/IIndexer.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
+import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
+import "@airswap/delegate/contracts/interfaces/IDelegate.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "@airswap/types/contracts/Types.sol";
+
+contract DelegateManager is Ownable {
+
+    event DelegateCreated(address owner, address delegate);
+
+    //keeps track of all the delegates created by owner address
+    mapping(address => address[]) private ownerToDelegates;
+    IDelegateFactory public factory;
+
+    /**
+      * @dev factory is unable to be changed after the DelegateManager is constructed
+      * @param _factory address of the factory contract that will create the delegates
+      */
+    constructor(IDelegateFactory _factory) public {
+        factory = _factory;
+    }
+
+    /**
+      * @notice Creates a new Delegate contract using the Factory contract, 
+      * saves deployed address to msg.sender list of deployed addresses
+      * @dev DelegateManager does not have access to the created Delegate by default
+      * @param _tradeWallet the wallet that the delegate will be trading on behalf of
+      * @return IDelegate the Delegate created by the factory
+      */
+    function createDelegate(address _tradeWallet) external returns (IDelegate) {
+      IDelegate delegate = IDelegate(factory.createDelegate(msg.sender, _tradeWallet));
+      ownerToDelegates[msg.sender].push(address(delegate));
+      emit DelegateCreated(msg.sender, address(delegate));
+      return delegate;
+    }
+
+    /**
+      * @notice gets all the delegates for an owner
+      * @param _owner the owner to look up addresses for
+      * @return address[] memory the list of the delegates for an owner
+      */ 
+    function getOwnerAddressToDelegates(address _owner) view external returns (address[] memory) {
+      uint256 length = ownerToDelegates[_owner].length;
+      address[] memory delegates = new address[](length);
+      for(uint i = 0; i < length; i++) {
+        delegates[i] = ownerToDelegates[_owner][i];
+      }
+      return delegates;
+    }
+
+    /**
+      * @notice sets a rule on the delegate and an intent on the indexer
+      * @dev delegate needs the manager to be an admin in order to act with it.
+      * @param _delegate the delegate that a rule will be set on
+      * @param _rule the rule to set on a delegate
+      * @param _intent the intent to set on an the indexer
+      * @param _indexer the indexer the intent will be set on
+      */
+    function setRuleAndIntent(
+      IDelegate _delegate,
+      Types.Rule calldata _rule,
+      Types.Intent calldata _intent,
+      IIndexer _indexer
+    ) external {
```

Comment:
feels like the caller of this method should be locked down to the owner of the delegate or something even if the delegatemanager is an admin to act on the delegate

Reply:
check the caller to see if they are the owner of the delegate.